tests/devices/unit_tests/test_bart_robot.py

```diff
@@ -80,3 +98,22 @@ async def test_given_program_not_running_and_pin_unmounts_then_mounts_when_load_
     assert (await device.next_puck.get_value()) == 15
     assert (await device.next_pin.get_value()) == 10
     device.load.trigger.assert_called_once()  # type:ignore
+
+
+async def test_given_waiting_for_pin_to_mount_when_no_pin_mounted_then_error_raised():
+    device = await _get_bart_robot()
+    status = device.pin_mounted_or_no_pin_found()
+    await sleep(0.01)
+    set_mock_value(device.error_code, 25)
+    await sleep(0.01)
+    with pytest.raises(RobotLoadFailed):
+        await status
+
```

Comment:
I don't understand what these sleeps are for - you don't enter the coroutine until `await` is called on it.